### PR TITLE
Fix: i18n type update

### PR DESCRIPTION
**Pull request by leightkt**
### Description
The documentation/discussion is in the linked issue https://github.com/blocknative/web3-onboard/issues/1632#issuecomment-1521869917 with context. Fixes Typescript error. 

### Checklist
- [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [x] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### If this PR includes changes to add an injected wallet or SDK wallet module: 
Please complete the following using the internal demo package.
To run this demo use the command `yarn && yarn dev` to get the project running at `http://localhost:8080/`

#### Tests with demo app (injected)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

#### Tests with demo app (SDK)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

### CODE REVIEW

File: packages/core/src/types.ts [hunk 1]
```diff
@@ -196,7 +196,11 @@ export type Configuration = {
 
 export type Locale = string
 export type i18nOptions = Record<Locale, i18n>
-export type i18n = typeof en
+// allows partial i18n override in TypeScript:
+type RecursivePartial<T> = { //https://stackoverflow.com/a/47914631
+  [P in keyof T]?: RecursivePartial<T[P]>;
```
Comment (Adamj1232) [inline packages/core/src/types.ts:201]:
```suggestion
type RecursivePartial<T> = { 
  // https://stackoverflow.com/a/47914631
  [P in keyof T]?: RecursivePartial<T[P]>;
```
Not sure if this link is even needed as it just points to stackoverflow, some explaining text would be more helpful

**Review by Adamj1232 — APPROVED** [review]
Approved but could either include documentation in place of the SO link or just remove all together

**Comment by leightkt** [comment]
@Adamj1232 I'll remove the link.

**Comment by wbt** [comment]
Thanks for including this!
While the StackOverflow link could be merged in with the longer comment, omitting it might be a violation of the license for using helpful content from StackOverflow (which requires attribution), so it might be a good idea to add that link back in.

**Pull request merged**